test/hotspot/gtest/runtime/test_os_windows.cpp

```diff
@@ -722,6 +722,105 @@ TEST_VM(os_windows, processor_count) {
+
+    do {
+      ++page_size_count;
+      page_size = os::page_sizes().next_smaller(page_size);
+    } while (page_size >= os::page_sizes().smallest());
+
+    EXPECT_GT(page_size_count, 1u) << "There should be multiple large page sizes available.";
+
+    size_t large_page_size = _large_page_size;
+    size_t min_size = os::page_sizes().smallest();
+
+    for (size_t page_size = os::page_sizes().largest(); page_size >= min_size; page_size = os::page_sizes().next_smaller(page_size)) {
+      EXPECT_TRUE(page_size % min_size == 0) << "Each page size should be a multiple of the minimum large page size.";
+      EXPECT_LE(page_size, large_page_size) << "Page size should not exceed the determined large page size.";
+    }
+  }
+#endif
+}
+
+TEST_VM(os_windows, large_page_init_decide_size) {
+  // Initial setup
+  UseLargePages = true;
+  bool isSupportedWindowsVersion = os::win32::is_windows_11_or_greater() ||
+    os::win32::is_windows_server_2022_or_greater();
+  LargePageSizeInBytes = 0; // Reset to default
+
+  // Test for large page support
+  size_t decided_size = os::large_page_init_decide_size();
+  size_t min_size = GetLargePageMinimum();
+  if (isSupportedWindowsVersion && min_size == 0) {
+    EXPECT_EQ(decided_size, 0) << "Expected decided size to be 0 when large page is not supported by the processor";
+  }
+
+  // Scenario 1: Test with 2MB large page size
+  LargePageSizeInBytes = 2 * M; // Set large page size to 2MB
+  decided_size = os::large_page_init_decide_size(); // Recalculate decided size
+  if (min_size == 2 * M) {
+    EXPECT_EQ(decided_size, 2 * M) << "Expected decided size to be 2M when large page and OS reported size are both 2M";
+  }
+
+  // Scenario 2: Test with 1MB large page size
+  LargePageSizeInBytes = 1 * M; // Set large page size to 1MB
+  decided_size = os::large_page_init_decide_size(); // Recalculate decided size
+  if (isSupportedWindowsVersion && min_size == 2 * M) {
+    EXPECT_EQ(decided_size, 2 * M) << "Expected decided size to be 2M when large page is 1M and OS reported size is 2M";
+  }
+
+#if defined(IA32) || defined(AMD64)
+  if (!isSupportedWindowsVersion) {
+    if (!EnableAllLargePageSizesForWindows && (min_size > 4 * M || LargePageSizeInBytes > 4 * M)) {
+      EXPECT_EQ(decided_size, 0) << "Expected decided size to be 0 for large pages bigger than 4mb on IA32 or AMD64";
+    }
+  }
+#endif
+
+  // Additional check for non-multiple of minimum size
+  // Set an arbitrary large page size which is not a multiple of min_size
+  LargePageSizeInBytes = 5 * min_size + 1;
+
+  // Recalculate decided size
+  decided_size = os::large_page_init_decide_size();
+
+  // Assert that the decided size defaults to minimum page size when LargePageSizeInBytes
+  // is not a multiple of the minimum size, assuming conditions are always met
+  EXPECT_EQ(decided_size, min_size) << "Expected decided size to default to minimum page size when LargePageSizeInBytes is not a multiple of minimum size";
```

Comment:
> Shouldn't this message use the "minimum large page size" instead of "minimum page size"? Also, last part should be "not a multiple of minimum large page size"

It defaults to the minimum page size because it fails to meet the conditions required to use a large page, specifically, it's not a multiple of the minimum large page size as determined by the getLargePageMinimum() function.

Reply:
> > Shouldn't this message use the "minimum large page size" instead of "minimum page size"? Also, last part should be "not a multiple of minimum large page size"
> 
> It defaults to the minimum page size because it fails to meet the conditions required to use a large page, specifically, it's not a multiple of the minimum large page size as determined by the getLargePageMinimum() function.

That part makes sense. The point I was trying to make here is that behavior you are describing is different from the behavior described in the error message.

Reply:
Also consider renaming min_size to large_page_min_size to make the code easier to understand.

Reply:
in the source code we have min_size for similar variable ...

Reply:
> in the source code we have min_size for similar variable ...

that doesn't mean that your test code can't use more descriptive names. you don't have to change this, it was just a suggestion - sometimes you can make code much clearer by using more descriptive variable names.

Reply:
Please fix the error message here though.

Reply:
This has been fixed and is outdated